BinaryEncoding.md

```diff
@@ -335,9 +332,9 @@ The `br_table` operator has an immediate operand which is encoded as follows:
 
 | Field | Type | Description |
 | ---- | ---- | ---- |
-| target_count | `uint16` | number of targets in the target_table |
-| target_table | `uint16*` | target entries that indicate an outer block or loop to which to break |
-| default_target | `uint16` | an outer block or loop to which to break in the default case |
+| target_count | `varuint32` | number of targets in the target_table |
+| target_table | `varuint32*` | target entries that indicate an outer block or loop to which to break |
```

Comment:
On Tue, Mar 8, 2016 at 8:04 PM, Luke Wagner <notifications@github.com>
wrote:

> In BinaryEncoding.md
> <https://github.com/WebAssembly/design/pull/592#discussion_r55409170>:
>
> > @@ -335,9 +332,9 @@ The `br_table` operator has an immediate operand which is encoded as follows:
> >
> >  | Field | Type | Description |
> >  | ---- | ---- | ---- |
> > -| target_count | `uint16` | number of targets in the target_table |
> > -| target_table | `uint16*` | target entries that indicate an outer block or loop to which to break |
> > -| default_target | `uint16` | an outer block or loop to which to break in the default case |
> > +| target_count | `varuint32` | number of targets in the target_table |
> > +| target_table | `varuint32*` | target entries that indicate an outer block or loop to which to break |
>
> I was going to make the argument that loop/block/if/else/end make direct
> interpretation infeasible, but the memoized jump table you've brought up
> would probably be effective. If we want to officially change our stance on
> direct interpretation from "infeasible, requires compilation to bytecode"
> to "not super-fast, but feasible, requires jump table" *and allow this to
> affect our design*, it'd be good to have a working proof of concept
> before the MVP (that has no separate bytecode, just side tables). Is this
> what you built with your debug interpreter? If so, cool!
>
Yes, I had that working for preorder, and it's even easier for postorder.
You just need a sidetable for if, else, br, br_if, and br_table opcodes.


> I would make an argument for size advantages of LEB128 over uint32 (given
> that the common case will be 1 byte in LEB128), but this is br_table
> which accounts for <.1% of total bytes, so I won't :)
>
> So uint32 seems like a fine compromise, I'll update now and we can
> reconsider later if necessary.
>
Ok, sgtm.


> —
> Reply to this email directly or view it on GitHub
> <https://github.com/WebAssembly/design/pull/592/files#r55409170>.
>